org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/contentmodel/XMLSchemaDiagnosticsTest.java

```diff
@@ -237,20 +248,73 @@ public void cvc_complex_type_2_4_bCodeAction_nonRootElement() throws Exception {
 		testDiagnosticsFor(xml, d);
 		testCodeActionsFor(xml, d, ca(d, te(2, 10, 2, 10, //
 				"\r\n" + //
-						"\t\t<author></author>\r\n\t")));
+						"\t\t<author></author>\r\n\t")),
+				ca(d, te(2, 10, 2, 10, //
+						"\r\n" + //
+								"\t\t<author></author>\r\n\t")));
 	}
 
 	@Test
-	public void cvc_complex_type_2_4_bCodeAction_withExistingChild() throws Exception {
+	public void cvc_complex_type_2_4_bCodeAction_Only_Required() throws Exception {
 		String xml = "<root xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"\r\n" + //
-				"    xsi:noNamespaceSchemaLocation=\"src/test/resources/xsd/unique_multiElement.xsd\">\r\n" + //
-				"    <authors><author></author></authors>\r\n" + //
+				"xsi:noNamespaceSchemaLocation=\"src/test/resources/xsd/unique_multiElement.xsd\"></root>";
+		Diagnostic d = d(0, 1, 0, 5, XMLSchemaErrorCode.cvc_complex_type_2_4_b);
+		testDiagnosticsFor(xml, d);
+		testCodeActionsFor(xml, d, ca(d, te(1, 79, 1, 79, //
+				"\r\n" + //
+						"\t<authors>\r\n" + //
+						"\t\t<author></author>\r\n" + //
+						"\t</authors>\r\n" + //
+						"\t<assistant></assistant>\n")),
+				ca(d, te(1, 79, 1, 79, //
+						"\r\n" + //
+								"\t<authors>\r\n" + //
+								"\t\t<author></author>\r\n" + //
+								"\t</authors>\n")));
+	}
+
+	@Test
+	public void cvc_complex_type_2_4_bCodeAction_Only_Required_Sequence() throws Exception {
+		String xml = "<root xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"\r\n" + //
+				"xsi:noNamespaceSchemaLocation=\"src/test/resources/xsd/order.xsd\"></root>";
+		Diagnostic d = d(0, 1, 0, 5, XMLSchemaErrorCode.cvc_complex_type_2_4_b);
+		testDiagnosticsFor(xml, d);
+		testCodeActionsFor(xml, d, ca(d, te(1, 65, 1, 65, //
+				"\r\n" + //
+						"\t<a></a>\r\n" + //
+						"\t<b></b>\r\n" + //
+						"\t<c></c>\r\n" + //
+						"\t<d age=\"\">\r\n" + //
+						"\t\t<d1></d1>\r\n" + //
+						"\t</d>\r\n" + //
+						"\t<e></e>\n")),
+				ca(d, te(1, 65, 1, 65, //
+						"\r\n" + //
+								"\t<a></a>\r\n" + //
+								"\t<d age=\"\"></d>\r\n" + //
+								"\t<e></e>\n")));
+	}
+
+	@Test
+	public void cvc_complex_type_2_4_bCodeAction_Only_Required_Ordered() throws Exception {
+		String xml = "<root xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"\r\n" + //
+				"xsi:noNamespaceSchemaLocation=\"src/test/resources/xsd/order.xsd\">\r\n" + //
+				"\t<a></a>\r\n" + //
+				"\t<c></c>\r\n" + //
 				"</root>";
 		Diagnostic d = d(0, 1, 0, 5, XMLSchemaErrorCode.cvc_complex_type_2_4_b);
 		testDiagnosticsFor(xml, d);
-		testCodeActionsFor(xml, d, ca(d, te(1, 83, 1, 83, //
+		testCodeActionsFor(xml, d, ca(d, te(1, 65, 1, 65, //
 				"\r\n" + //
-						"\t<assistant></assistant>")));
+						"\t<b></b>\r\n" + //
+						"\t<d age=\"\">\r\n" + //
+						"\t\t<d1></d1>\r\n" + //
+						"\t</d>\r\n" + //
+						"\t<e></e>")),

```

Comment:
The new elements current insert right after the parent and is not in the correct order. I believe it would be a good idea to fix this in another PR in relation to this issue: https://github.com/eclipse/lemminx/issues/1256.

Reply:
Indeed let's go step by step. Please create an issue for that.